apps/backend/modules/runner.py

```diff
@@ -5,70 +5,60 @@
 import os
 
 # from modules.data.trial import Trial
-from modules.configs import get_configs_ordered
-from modules.exceptions import ExperimentAbort, FileHandlingError, GladosUserError, TrialTimeoutError
+from modules.configs import create_config_from_data, get_configs_ordered
+from modules.data.experiment import ExperimentData, ExperimentType
+from modules.exceptions import ExperimentAbort, FileHandlingError, GladosInternalError, GladosUserError, TrialTimeoutError
 from modules.exceptions import InternalTrialFailedError
 from modules.configs import get_config_paramNames
 
-PIPE_OUTPUT_ERROR_MESSAGE = "ERROR"
-OUTPUT_INDICATOR_USING_CSVS = "In ResCsvs"
+PROCESS_OUT_STREAM = 0
+PROCESS_ERROR_STREAM = 1
 
 
 def get_data(process: 'Popen[str]', trialRun: int, keepLogs: bool, trialTimeout: int):
     try:
         data = process.communicate(timeout=trialTimeout)
         if keepLogs:
             os.chdir('ResCsvs')
-            with open(f"log{trialRun}.txt", 'w', encoding='utf8') as f:
-                f.write(data[0])
+            with open(f"log{trialRun}.txt", 'w', encoding='utf8') as trialLogFile:
+                trialLogFile.write(data[PROCESS_OUT_STREAM])
                 if data[1]:
-                    f.write(data[1])
-                f.close()
+                    trialLogFile.write(data[PROCESS_ERROR_STREAM])
+                trialLogFile.close()
             os.chdir('..')
-        if data[1]:
-            print(f'errors returned from pipe is {data[1]}')
-            return PIPE_OUTPUT_ERROR_MESSAGE
-    except TimeoutExpired as timeErr: 
+        if data[PROCESS_ERROR_STREAM]:
+            errorMessage = f'errors returned from pipe is {data[PROCESS_ERROR_STREAM]}'
+            print(errorMessage)
+            raise InternalTrialFailedError(errorMessage)
+    except TimeoutExpired as timeErr:
```

Comment:
can we change this to use exceptions instead of the error flags?

Reply:
It does use exceptions, the Stream constants are to get the data out of the popen, since the sysout and errors are in different areas